v4/core/utils.go

```diff
@@ -179,3 +179,57 @@ func PrettyPrint(result interface{}, resultName string) {
 func GetCurrentTime() int64 {
 	return time.Now().Unix()
 }
+
+// ConvertSlice Marshals 'slice' to a json string, performs
+// string manipulation on the resulting string, and converts
+// the string to a '[]string'. If 'slice' is nil, not a 'slice' type,
+// or an error occurred during conversion, an empty slice will be returned.
+func ConvertSlice(slice interface{}) (s []string) {
+	s = []string{}
+	inputIsSlice := false
+
+	if IsNil(slice) {
+		return
+	}
+
+	// Reflect on 'slice' to validate the input is in fact a slice
+	rResultType := reflect.TypeOf(slice)
+
+	switch rResultType.Kind() {
+	case reflect.Slice:
+		inputIsSlice = true
+	default:
+	}
+
+	// If it's not a slice or if it's an empty slice, just return
+	if !inputIsSlice || reflect.ValueOf(slice).Len() == 0 {
+		return
+	}
+
+	jsonBuffer, err := json.Marshal(slice)
+	if err != nil {
+		return
+	}
+
+	// Convert the json buffer to a string and remove any escaped quotes
+	jsonString := strings.Replace(string(jsonBuffer), "\"", "", -1)
+	jsonString = strings.Replace(jsonString, "\\", "", -1)
+
+	// Use regex to convert the json string to a string slice
+	re := regexp.MustCompile(`(?s)\[(\S*)\]`)
```

Comment:
A simple regex that validates that the returned string is in the format: "[(any character(s))]"